src/amici.cpp

```diff
@@ -64,6 +64,19 @@ void runAmiciSimulation(UserData *udata, const ExpData *edata, ReturnData *rdata
     applyChainRuleFactorToSimulationResults(udata, rdata, edata);
 
 freturn:
+    if(*pstatus<0){
+        if(llhdata)
+            *llhdata = amiGetNaN();
+        
+        if(sllhdata)
+            for (int ip=0; ip<nplist; ip++)
+            sllhdata[ip] = amiGetNaN();
```

Comment:
or put it to a separate function 
void invalidateReturnData(ReturnData*, UserData*);